hs_core/views/__init__.py

```diff
@@ -479,74 +515,34 @@ class AddGroupForm(forms.Form):
-    ret_responses = pre_describe_resource.send(sender=res_cls, files=resource_files, title=res_title)
-
-    create_res_context = {
-        'resource_type': resource_type,
-        'res_title': res_title,
-    }
-    page_url = 'pages/create-resource.html'
-    use_generic = True
-    for receiver, response in ret_responses:
-        if response is not None:
-            for key in response:
-                if key != 'create_resource_page_url':
-                    create_res_context[key] = response[key]
-                else:
-                    page_url = response.get('create_resource_page_url', 'pages/create-resource.html')
-                    use_generic = False
-
-    if use_generic:
-        # create barebone resource with resource_files to database model for later update since on Django 1.7, resource_files get closed automatically at the end of each request
-        owner = user_from_id(request.user)
-        resource = res_cls.objects.create(
-                user=owner,
-                creator=owner,
-                title=res_title,
-                last_changed_by=owner,
-                in_menus=[],
-                **kwargs
-        )
-        for file in resource_files:
-            ResourceFile.objects.create(content_object=resource, resource_file=file)
-
-    return render_to_response(page_url, create_res_context, context_instance=RequestContext(request))
-
 
 @login_required
 def create_resource_select_resource_type(request, *args, **kwargs):
     return render_to_response('pages/create-resource.html', context_instance=RequestContext(request))
 
-
-class CreateResourceForm(forms.Form):
-    title = forms.CharField(required=True)
-    creators = forms.CharField(required=False, min_length=0)
-    contributors = forms.CharField(required=False, min_length=0)
-    abstract = forms.CharField(required=False, min_length=0)
-    keywords = forms.CharField(required=False, min_length=0)
-
 @login_required
 def create_resource(request, *args, **kwargs):
     resource_type = request.POST['resource-type']
     res_title = request.POST['title']
 
-    url_key = "page_redirect_url"
     resource_files = request.FILES.getlist('files')
 
+    irods_fname = request.session.get('irods_file_name', '')
+    if irods_fname:
+        user = request.session["user"]
+        password = request.session["password"]
+        port = request.session["port"]
+        host = request.session["host"]
+        zone = request.session["zone"]
+        # use iget to transfer selected data object to local as a NamedTemporaryFile
+        irods_storage = IrodsStorage()
+        irods_storage.set_user_session(username=user, password=password, host=host, port=port, zone=zone)
+        tmpFile = irods_storage.download(irods_fname)
+        fname = os.path.basename(irods_fname.rstrip(os.sep))
+        resource_files.append(UploadedFile(file=tmpFile, name=fname))
```

Comment:
same code as above in the add view. Can it be put in a utility method somewhere?